flow/connectors/bigquery/merge_stmt_generator.go

```diff
@@ -100,6 +101,32 @@ func (m *mergeStmtGenerator) generateFlattenedCTE() string {
 		m.syncBatchID, m.dstTableName)
 }
 
+// This function is to support datatypes like JSON which cannot be compared by BigQuery
+func (m *mergeStmtGenerator) transformedPkeyStrings(forPartition bool) []string {
+	pkeys := make([]string, 0, len(m.normalizedTableSchema.PrimaryKeyColumns))
+	for _, col := range m.normalizedTableSchema.Columns {
+		if slices.Contains(m.normalizedTableSchema.PrimaryKeyColumns, col.Name) {
+			pkeyCol := col.Name
+			switch qvalue.QValueKind(col.Type) {
+			case qvalue.QValueKindJSON:
```

Comment:
I think we also need to do this with floats for partition by

GEOGRAPHY/STRUCT/ARRAY can be in a followup PR